lib/graphql/pagination/active_record_relation_connection.rb

```diff
@@ -5,6 +5,26 @@ module GraphQL
   module Pagination
     # Customizes `RelationConnection` to work with `ActiveRecord::Relation`s.
     class ActiveRecordRelationConnection < Pagination::RelationConnection
+      def has_next_page
+        if @has_next_page.nil?
+          @has_next_page = if before_offset && before_offset > 0
+            true
+          elsif first
+            if @nodes && @nodes.count < first
+              false
+            else
+              initial_offset = sliced_nodes.offset_value || 0
+              sliced_nodes.offset(initial_offset + first).exists?
+            end
+          else
+            false
+          end
+        end
+        @has_next_page
+      end
```

Comment:
I pretty much just copied over the `has_next_page` method from RelationConnection and changed the relevant part. We could remove some duplication between the two classes:

```ruby
def has_next_page
  return @has_next_page unless @has_next_page.nil?
  return super if before_offset && before_offset > 0 || !first

  @has_next_page = if @nodes && @nodes.count < first
    false
  else
    initial_offset = sliced_nodes.offset_value || 0
    sliced_nodes.offset(initial_offset + first).exists?
  end
end
```

This would make it harder to understand the method, though.

Reply:
I moved the check into a private method and am now just overriding that in ActiveRecordRelationConnection.